File: submissions/KRISHNA GUPTA/Day28_sol.py
text=input("Enter cipher text:")
n=int(input("Enter shift value:"))
decoded_message=" "
for char in text:
    if char.isalpha():
        base = ord('A') if char.isupper() else ord('a')
        decoded_char = chr((ord(char) - base - n) % 26 + base)
        decoded_message += decoded_char
    else:
        decoded_message += char

print("Decoded message:", decoded_message)

File: submissions/Abhishek_Seth/Day_28_sol/Day_28_sol.java
import java.util.*;
public class Day_28_sol{
    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        System.out.print("Enter cipher text: ");
        String s2="";
        String s1= sc.nextLine();
        System.out.print("Enter shift value: ");
        int n=sc.nextInt();
        System.out.print("Decoded Message: ");
        for (int i = 0; i < s1.length(); i++) {
            char w = s1.charAt(i);
            if(Character.isLetter(w)){
                if(Character.isUpperCase(w)){
                    w = (char) ('A' + ((w - 'A' - n + 26) % 26));
                }
                else if(Character.isLowerCase(w)){
                    w = (char) ('a' + ((w - 'a' - n + 26) % 26));
                }
            }
            System.out.print(w);
        }
        
    }
}